### PR TITLE
merging and fixing issue #13

### PR DESCRIPTION
**Pull request by jdr0887**
Removes dependency on horned_bin

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,10 +5,11 @@ authors = ["Jim Balhoff <balhoff@renci.org>"]
 edition = "2021"
 
 [dependencies]
+curie = "^0.1"
 clap = { version = "^4.1", features = ["derive"] }
 env_logger = "^0.10"
 #horned-functional = "^0.4"
-horned-bin = "^0.14"
+#horned-bin = "^0.14"
```
Comment (balhoff) [inline Cargo.toml]: Just delete instead of commenting.

File: src/main.rs [hunk 2]
```diff
@@ -57,3 +61,21 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
+    match input_path.extension().and_then(|s| s.to_str()) {
+        Some("owx") => {
+            let ret = horned_owl::io::owx::reader::read(&mut bufreader, config).expect("unable to parse input");
+            Ok(ret.0)
+        }
+        Some("owl") => {
+            let ret = horned_owl::io::rdf::reader::read(&mut bufreader, config).expect("unable to parse input");
+            Ok(ret.0.into())
+        }
+        _ => Err(Box::<dyn error::Error>::from("unable to parse input")),
```
Comment (balhoff) [inline src/main.rs]: Why not return the error from horned-owl? Could it provide useful information that could be logged about why parsing failed?

**Review by balhoff — CHANGES_REQUESTED** [review]
Looks good! I had a question about logging parse errors.

File: src/main.rs (hunk 2 above)
Comment (jdr0887) [reply src/main.rs]: The leafs of the match are different.  I will change the .expect() to ? so that any error will trickle up.

Comment (balhoff) [reply src/main.rs]: Okay, thanks.

**Review by balhoff — APPROVED** [review]

**Pull request merged**